avocado/utils/process.py

```diff
@@ -267,16 +267,32 @@ class CmdResult(object):
     :param pid: ID of the process
     """
 
-    def __init__(self, command="", stdout="", stderr="",
+    def __init__(self, command="", stdout_bytes=b"", stderr_bytes=b"",
                  exit_status=None, duration=0, pid=None):
         self.command = command
         self.exit_status = exit_status
-        self.stdout = stdout
-        self.stderr = stderr
+        self.stdout_bytes = stdout_bytes
+        self.stderr_bytes = stderr_bytes
+        self._stdout = None
+        self._stderr = None
         self.duration = duration
         self.interrupted = False
         self.pid = pid
 
+    @property
+    def stdout(self):
+        if self._stdout is not None:
+            return self._stdout
+        self._stdout = self.stdout_bytes.decode('utf-8')
```

Comment:
Hello @bonzini, thank you for all the python3 related patches. This one troubles me quite a lot as we've had bugs reported by users with non-utf-8 encodings, so `decode('utf-8')` is not an answer everywhere. The thing is I don't really know the solution, we used combination of PYTHONENCODING, PYTHONIOENCODING and `sys.getdefaultencoding()` and changing `ascii` with `utf8` (as python2 reports most frequently ascii).

Ideal solution would be to include some tests for this module for encoding together with this patch, some could be inspired by https://github.com/python/cpython/blob/master/Lib/test/test_subprocess.py

PS: The `avocado.utils.process` is heavily utilized module, that's why I'm so concerned about basically any change here.